pkg/source/source.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package source
+
+import (
+	"context"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/mattmoor/mink/pkg/kontext"
+	tknv1beta1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// GitLocation allows cloning from git
+type GitLocation struct {
+	// URL the git repo URL
+	URL string
+
+	// Revision is the git revision to clone
+	Revision string
+
+	// Verbose enable verbose logging for git
+	Verbose bool
+}
+
+var (
+	// based on a simplification of the git clone task from the tekton catalog:
+	// https://github.com/tektoncd/catalog/tree/master/task/git-clone
+	gitCloneImage = "gcr.io/tekton-releases/github.com/tektoncd/pipeline/cmd/git-init:v0.18.0"
```

Comment:
We should be able to build this from source out of `vendor` rather than relying on the pre-built images.